src/Component/DOMElementList.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Class DOMElementList
+ *
+ * @package Amp\AmpWP
+ */
+
+namespace Amp\AmpWP\Component;
+
+use IteratorAggregate;
+use Countable;
+use DOMElement;
+use ArrayIterator;
+
+/**
+ * Class DOMElementList
+ *
+ * @internal
+ * @since 1.5.0
+ */
+final class DOMElementList implements IteratorAggregate, Countable {
+
+	/**
+	 * The elements, possibly with captions.
+	 *
+	 * @var DOMElement[]
+	 */
+	private $elements = [];
+
+	/**
+	 * Adds an image to the list.
+	 *
+	 * @param DOMElement $element The element to add, possibly an image.
+	 * @param string     $caption The caption to add.
+	 * @return self
+	 */
+	public function add( DOMElement $element, $caption = '' ) {
+		$this->elements[] = empty( $caption ) ? $element->cloneNode() : new CaptionedSlide( $element->cloneNode(), $caption );
```

Comment:
Shouldn't this be a deep clone? Otherwise the `noscript > img` children won't be included.

```suggestion
		$this->elements[] = empty( $caption ) ? $element->cloneNode( true ) : new CaptionedSlide( $element->cloneNode( true ), $caption );
```

Reply:
Yes, good idea.

Reply:
That helped, but the tests still failed locally. I'm working on it.